### PR TITLE
ekf2: mag_fusion consolidate duplicate error handling

### PR DESCRIPTION
**Pull request by dagar**

### CODE REVIEW

**Review by bresch — DISMISSED** [review]

File: src/modules/ekf2/EKF/mag_fusion.cpp [hunk 1]
```diff
@@ -211,11 +177,14 @@ bool Ekf::fuseMag(const Vector3f &mag, estimator_aid_source3d_s &aid_src_mag, bo
 		VectorState Kfusion = P * H / aid_src_mag.innovation_variance[index];
 
 		if (!update_all_states) {
-			for (unsigned row = 0; row <= 15; row++) {
+			unsigned mag_index_first = math::min(State::mag_I.idx, State::mag_B.idx);
```
Comment (bresch) [inline src/modules/ekf2/EKF/mag_fusion.cpp:180]: Alternatively we could also do it like here: https://github.com/PX4/PX4-Autopilot/blob/5f87f3a04652d74ea21f3d39ec687f287a7da45b/src/modules/ekf2/EKF/airspeed_fusion.cpp#L201-L203
to make sure we never assume that mag_I and mag_B are next to each other

**Comment by dagar** [comment]
> > other parts of the system are blanket checking for any fault status flag
> 
> Commander and EKF2Selector?

EKF2Selector and ekf2 sensor calibration.

https://github.com/PX4/PX4-Autopilot/blob/20644dd5fc1b6e7f9870a3f50db7916915e69515/src/modules/ekf2/EKF2Selector.cpp#L302C40-L302C58
https://github.com/PX4/PX4-Autopilot/blob/20644dd5fc1b6e7f9870a3f50db7916915e69515/src/modules/ekf2/EKF2.cpp#L2662

File: src/modules/ekf2/EKF/mag_fusion.cpp (hunk 1 above)
Comment (dagar) [reply src/modules/ekf2/EKF/mag_fusion.cpp:180]: Yes that's probably better.

Comment (dagar) [reply src/modules/ekf2/EKF/mag_fusion.cpp:180]: updated

**Pull request merged**